### PR TITLE
Add RHEL 8 rules from PowerTools repo

### PR DESCRIPTION
**Pull request by cottsay**
This repo is published by RedHat, but it is not enabled by default. It contains quite a few packages that were previously available in the 'base' or 'epel' repos for RHEL 7.

I also explicitly listed the AppStream repo in CONTRIBUTING.md. This repo is required for a basic installation (similar to BaseOS), but I thought it best to list it since we list 'base'.

There aren't any official web databases for RHEL/CentOS packages, so I'll link to the packages sitting in the repository mirrors.

| rosdep key | system package |
|------------|----------------|
| boost | [boost-python%{python3_pkgversion}-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/boost-python3-devel-1.66.0-6.el8.x86_64.rpm) |
| graphviz-dev | [graphviz-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/graphviz-devel-2.40.1-39.el8.x86_64.rpm) |
| jasper | [jasper-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/jasper-devel-2.0.14-4.el8.x86_64.rpm) |
| libavahi-client-dev | [avahi-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/avahi-devel-0.7-19.el8.x86_64.rpm) |
| libavahi-core-dev | [avahi-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/avahi-devel-0.7-19.el8.x86_64.rpm) |
| libbison-dev | [bison-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/bison-devel-3.0.4-10.el8.x86_64.rpm) |
| libblas-dev | [blas-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/blas-devel-3.8.0-8.el8.x86_64.rpm) |
| libbluetooth-dev | [bluez-libs-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/bluez-libs-devel-5.50-1.el8.x86_64.rpm) |
| libconfig-dev | [libconfig-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libconfig-devel-1.5-9.el8.x86_64.rpm) |
| libfltk-dev | [fltk-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/fltk-devel-1.3.4-5.el8.x86_64.rpm) |
| libgflags-dev | [gflags-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/gflags-devel-2.1.2-6.el8.x86_64.rpm) |
| libglew-dev | [glew-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/glew-devel-2.0.0-6.el8.x86_64.rpm) |
| libglm-dev | [glm-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/glm-devel-0.9.8.5-2.el8.noarch.rpm) |
| libgoogle-glog-dev | [glog-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/glog-devel-0.3.5-3.el8.x86_64.rpm) |
| libgphoto-dev | [libgphoto2-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libgphoto2-devel-2.5.16-3.el8.x86_64.rpm) |
| libjson0-dev | [json-c-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/json-c-devel-0.13.1-0.2.el8.x86_64.rpm) |
| liblapack-dev | [lapack-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/lapack-devel-3.8.0-8.el8.x86_64.rpm) |
| liblttng-ust-dev | [lttng-ust-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/lttng-ust-devel-2.8.1-11.el8.x86_64.rpm) |
| libmicrohttpd | [libmicrohttpd-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libmicrohttpd-devel-0.9.59-2.el8.x86_64.rpm) |
| libmp3lame-dev | [lame-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/lame-devel-3.100-6.el8.x86_64.rpm) |
| libopenal-dev | [openal-soft-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/openal-soft-devel-1.18.2-7.el8.x86_64.rpm) |
| libopenblas-dev | [openblas-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/openblas-devel-0.3.3-2.el8.x86_64.rpm) |
| libopenexr-dev | [OpenEXR-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/OpenEXR-devel-2.2.0-11.el8.x86_64.rpm) |
| libpcap | [libpcap-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libpcap-devel-1.9.0-3.el8.x86_64.rpm) |
| libqhull | [qhull-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/qhull-devel-2015.2-5.el8.x86_64.rpm) |
| librabbitmq-dev | [librabbitmq-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/librabbitmq-devel-0.9.0-1.el8.x86_64.rpm) |
| libraw1394-dev | [libraw1394-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libraw1394-devel-2.1.2-5.el8.x86_64.rpm) |
| libsndfile1-dev | [libsndfile-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libsndfile-devel-1.0.28-8.el8.x86_64.rpm) |
| libturbojpeg | [turbojpeg-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/turbojpeg-devel-1.5.3-10.el8.x86_64.rpm) |
| libusb-dev | [libusb-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libusb-devel-0.1.5-12.el8.x86_64.rpm) |
| libv4l-dev | [libv4l-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libv4l-devel-1.14.2-3.el8.x86_64.rpm) |
| libvpx-dev | [libvpx-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/libvpx-devel-1.7.0-6.el8.x86_64.rpm) |
| libxmlrpc-c++ | [xmlrpc-c-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/xmlrpc-c-devel-1.51.0-5.el8.x86_64.rpm) |
| nasm | [nasm](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/nasm-2.13.03-2.el8.x86_64.rpm) |
| netpbm | [netpbm-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/netpbm-devel-10.82.00-6.el8.x86_64.rpm) |
| nkf | [nkf](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/nkf-2.1.4-8.el8.x86_64.rpm) |
| opencl-headers | [opencl-headers](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/opencl-headers-2.2-1.20180306gite986688.el8.noarch.rpm) |
| pandoc | [pandoc](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/pandoc-2.0.6-4.el8.x86_64.rpm) |
| protobuf-dev | [protobuf-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/protobuf-devel-3.5.0-7.el8.x86_64.rpm) |
| protobuf-dev | [protobuf-compiler](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/protobuf-compiler-3.5.0-7.el8.x86_64.rpm) |
| pstoedit | [pstoedit](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/pstoedit-3.70-9.el8.x86_64.rpm) |
| sdl2 | [SDL2-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/SDL2-devel-2.0.8-7.el8.x86_64.rpm) |
| snappy | [snappy-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/snappy-devel-1.1.7-5.el8.x86_64.rpm) |
| suitesparse | [suitesparse-devel](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/suitesparse-devel-4.4.6-11.el8.x86_64.rpm) |
| python3-packaging | [python%{python3_pkgversion}-packaging](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/python3-packaging-16.8-9.el8.noarch.rpm) |
| python3-sphinx | [python%{python3_pkgversion}-sphinx](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/python3-sphinx-1.7.6-1.el8.noarch.rpm) |
| python3-sphinx-rtd-theme | [python%{python3_pkgversion}-sphinx_rtd_theme](http://mirror.centos.org/centos-8/8.1.1911/PowerTools/x86_64/os/Packages/python3-sphinx_rtd_theme-0.3.1-3.el8.noarch.rpm) |

```
$ dnf list -q --showduplicates --available boost-python3-devel \
> graphviz-devel jasper-devel avahi-devel bison-devel blas-devel \
> bluez-libs-devel libconfig-devel fltk-devel gflags-devel glew-devel \
> glm-devel glog-devel libgphoto2-devel json-c-devel lapack-devel \
> lttng-ust-devel libmicrohttpd-devel lame-devel openal-soft-devel \
> openblas-devel OpenEXR-devel libpcap-devel qhull-devel librabbitmq-devel \
> libraw1394-devel libsndfile-devel turbojpeg-devel libusb-devel \
> libv4l-devel libvpx-devel xmlrpc-c-devel nasm netpbm-devel nkf \
> opencl-headers pandoc protobuf-devel protobuf-compiler pstoedit SDL2-devel \
> snappy-devel suitesparse-devel python3-packaging python3-sphinx \
> python3-sphinx_rtd_theme --exclude *.i686
Available Packages
OpenEXR-devel.x86_64                2.2.0-11.el8                     PowerTools
SDL2-devel.x86_64                   2.0.8-7.el8                      PowerTools
avahi-devel.x86_64                  0.7-19.el8                       PowerTools
bison-devel.x86_64                  3.0.4-10.el8                     PowerTools
blas-devel.x86_64                   3.8.0-8.el8                      PowerTools
bluez-libs-devel.x86_64             5.50-1.el8                       PowerTools
boost-python3-devel.x86_64          1.66.0-6.el8                     PowerTools
fltk-devel.x86_64                   1.3.4-5.el8                      PowerTools
gflags-devel.x86_64                 2.1.2-6.el8                      PowerTools
glew-devel.x86_64                   2.0.0-6.el8                      PowerTools
glm-devel.noarch                    0.9.8.5-2.el8                    PowerTools
glog-devel.x86_64                   0.3.5-3.el8                      PowerTools
graphviz-devel.x86_64               2.40.1-39.el8                    PowerTools
jasper-devel.x86_64                 2.0.14-4.el8                     PowerTools
json-c-devel.x86_64                 0.13.1-0.2.el8                   PowerTools
lame-devel.x86_64                   3.100-6.el8                      PowerTools
lapack-devel.x86_64                 3.8.0-8.el8                      PowerTools
libconfig-devel.x86_64              1.5-9.el8                        PowerTools
libgphoto2-devel.x86_64             2.5.16-3.el8                     PowerTools
libmicrohttpd-devel.x86_64          1:0.9.59-2.el8                   PowerTools
libpcap-devel.x86_64                14:1.9.0-3.el8                   PowerTools
librabbitmq-devel.x86_64            0.9.0-1.el8                      PowerTools
libraw1394-devel.x86_64             2.1.2-5.el8                      PowerTools
libsndfile-devel.x86_64             1.0.28-8.el8                     PowerTools
libusb-devel.x86_64                 1:0.1.5-12.el8                   PowerTools
libv4l-devel.x86_64                 1.14.2-3.el8                     PowerTools
libvpx-devel.x86_64                 1.7.0-6.el8                      PowerTools
lttng-ust-devel.x86_64              2.8.1-11.el8                     PowerTools
nasm.x86_64                         2.13.03-2.el8                    PowerTools
netpbm-devel.x86_64                 10.82.00-6.el8                   PowerTools
nkf.x86_64                          1:2.1.4-8.el8                    PowerTools
openal-soft-devel.x86_64            1.18.2-7.el8                     PowerTools
openblas-devel.x86_64               0.3.3-2.el8                      PowerTools
opencl-headers.noarch               2.2-1.20180306gite986688.el8     PowerTools
pandoc.x86_64                       2.0.6-4.el8                      PowerTools
protobuf-compiler.x86_64            3.5.0-7.el8                      PowerTools
protobuf-devel.x86_64               3.5.0-7.el8                      PowerTools
pstoedit.x86_64                     3.70-9.el8                       PowerTools
python3-packaging.noarch            16.8-9.el8                       PowerTools
python3-sphinx.noarch               1:1.7.6-1.el8                    PowerTools
python3-sphinx_rtd_theme.noarch     0.3.1-3.el8                      PowerTools
qhull-devel.x86_64                  2015.2-5.el8                     PowerTools
snappy-devel.x86_64                 1.1.7-5.el8                      PowerTools
suitesparse-devel.x86_64            4.4.6-11.el8                     PowerTools
turbojpeg-devel.x86_64              1.5.3-10.el8                     PowerTools
xmlrpc-c-devel.x86_64               1.51.0-5.el8                     PowerTools
```

### CODE REVIEW

**Review by dirk-thomas — APPROVED** [review]

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2659,6 +2669,7 @@ liblttng-ust-dev:
   debian: [liblttng-ust-dev]
   fedora: [lttng-ust]
```
Comment (sloretz) [inline rosdep/base.yaml:2670]: Should fedora be `lttng-ust-devel` here too?

File: rosdep/base.yaml [hunk 2]
```diff
@@ -4832,6 +4856,7 @@ osmium:
 pandoc:
```
Comment (sloretz) [inline rosdep/base.yaml:4856]: pandoc need a fedora rule too?

File: rosdep/base.yaml [hunk 3]
```diff
@@ -5395,6 +5421,7 @@ snappy:
   debian: [libsnappy-dev]
   fedora: [snappy]
```
Comment (sloretz) [inline rosdep/base.yaml:5422]: Should fedora be `snappy-devel` too?

**Review by sloretz — APPROVED** [review]
LGTM, had some questions about existing fedora rules

**Pull request merged**